Assets/Scripts/StandardSamples/Services/AchievementsService.cs

```diff
@@ -109,7 +109,11 @@ private static string ProductUserIdToString(ProductUserId productUserId)
 
         protected async override void OnLoggedIn(AuthenticationListener.LoginChangeKind changeType)
         {
-            await RefreshAsync();
+            // Achievements require Epic Game Services to function, which requires a Connect login
+            if (changeType == AuthenticationListener.LoginChangeKind.Connect)
+            {
+                await RefreshAsync();
+            }
```

Comment:
The implementation of `RefreshAsync()` is as follows:

```csharp
// Check to see if authentication is required, if it's not then 
// continue. If it is, then make sure a user is authenticated before
// refreshing.
if (!RequiresAuthentication || 
    (RequiresAuthentication && AuthenticationListener.Instance.IsAuthenticated))
{
    await InternalRefreshAsync();
}
```

So it seems incorrect to make a double check on this - is there a reason that the check that's done inside the `RefreshAsync` function is insufficient? If so - we should figure out how to resolve the issue so that the conditional already within the base implementation of `RefreshAsync` resolves the way we expect it to.

If the conditional _within_ `RefreshAsync` is not behaving properly, that indicates that there is a bug within the value being set for the property `RequiresAuthentication`, or within the value being set for the property `AuthenticationListener.Instance.IsAuthenticated`.

Reply:
Wouldn't the refresh in the RefreshAsync also fire if _only_ a EGS aka auth login had occurred?

Reply:
The refresh in the `RefreshAsync` base implementation will only fire if an EGS aka auth login occurred _AND_ such an auth login is required for the service in question.

Reply:
I am working on a fix that clarifies the distinct roles of Auth Interface authentication vs Connect Interface authentication - as the current implementation's limitation of having a `boolean` value that simply determines if _any_ authentication is the crux of the issue that this PR seeks to resolve. 

I've coordinated with @WispyMouse regarding how to proceed with these changes on their behalf. To be clear - the changes proposed in this PR as they currently exist are insufficient to address the problem and serve to compound an already incorrect implementation.